arch/arm/src/nrf52/nrf52_i2c.c

```diff
@@ -68,6 +68,11 @@ struct nrf52_i2c_priv_s
   uint8_t                 msgc;     /* Message count */
   struct i2c_msg_s       *msgv;     /* Message list */
   uint8_t                *ptr;      /* Current message buffer */
+#ifdef CONFIG_NRF52_I2C_MASTER_COPY_BUF_SIZE
+  /* Static buffer used for continued messages */
+
+  uint8_t                 copy_buf[CONFIG_NRF52_I2C_MASTER_COPY_BUF_SIZE];
```

Comment:
why not change copy_buf to pointer and extend the buffer on the fly?

Reply:
The whole point is to not have to do the allocate and free for small transfers. This provides a large performance increase.

Reply:
But , why not avoid the allocaiton every time for the large transfer?
```
struct nrf52_i2c_priv_s
{
...
  uint8_t *copy_buf;
  size_t    copy_len;
...
};

static int nrf52_i2c_transfer(FAR struct i2c_master_s *dev,
                              FAR struct i2c_msg_s *msgs,
                              int count)
{
                  if ((priv->msgv[0].length + priv->msgv[1].length) <= priv->copy_len)
                    {
                       pack_buf  = priv->copy_buf;
                    }
                  else
                   {
                      pack_buf = kmm_realloc(priv->copy_buf, priv->msgv[0].length + priv->msgv[1].length);
                      if (pack_buf == NULL)
                        {
                          return -ENOMEM;
                        }
                      priv->copy_buf = pack_buf;
                      priv->copy_len = priv->msgv[0].length + priv->msgv[1].length;
                   }
```

Reply:
> But , why not avoid the allocaiton every time for the large transfer?

I thought about doing realloc but I'm worried about leaving these possible large allocations that cannot be freed which may have only been used to write a large buffer to an eeprom.